pkg/cloud/services/ssm/parameter.go

```diff
@@ -0,0 +1,73 @@
+/*
+Copyright 2020 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package ssm
+
+import (
+	"strings"
+
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/aws/request"
+	"github.com/aws/aws-sdk-go/service/ssm"
+	"sigs.k8s.io/cluster-api-provider-aws/pkg/cloud/scope"
+)
+
+const (
+	parameterPrefix  = "cluster-api-provider-aws.sigs.k8s.io"
+	namespacesPrefix = "namespaces"
+	clustersPrefix   = "clusters"
+	instancesPrefix  = "instances"
+)
+
+func (s *Service) parameterName(m *scope.MachineScope) string {
+	return strings.Join(
+		[]string{
+			parameterPrefix,
+			namespacesPrefix,
+			s.scope.Namespace(),
+			clustersPrefix,
+			s.scope.Name(),
+			instancesPrefix,
+			m.Name(),
+		},
+		"/",
+	)
+}
+
+func (s *Service) putParameter(m *scope.MachineScope, data string) error {
+	_, err := s.scope.SSM.PutParameter(&ssm.PutParameterInput{
+		Name:      aws.String(s.parameterName(m)),
+		Value:     aws.String(data),
+		Type:      aws.String(ssm.ParameterTypeSecureString),
+		Overwrite: aws.Bool(true),
+	})
+	return err
+}
+
+func (s *Service) constructGetParameterRequest(m *scope.MachineScope) *request.Request {
+	req, _ := s.scope.SSM.GetParameterRequest(&ssm.GetParameterInput{
+		Name:           aws.String(s.parameterName(m)),
+		WithDecryption: aws.Bool(true),
+	})
+	return req
+}
+
+func (s *Service) constructDeleteParameterRequest(m *scope.MachineScope) *request.Request {
+	req, _ := s.scope.SSM.DeleteParameterRequest(&ssm.DeleteParameterInput{
```

Comment:
Is this actually needed? It seems straightforward enough to not be in its own method.

Also why are we ignoring the error?